Remove deprecated logic and emit clear messages (#2211)
jupyterhub/zero-to-jupyterhub-k8s: jupyterhub/templates/NOTES.txt

######   WARNING: You are using user placeholders without pod priority      #####
######            enabled*, either enable pod priority or stop using the    #####
######            user placeholders** to avoid having placeholders that     #####
######            refuse to make room for a real user.                      #####
######                                                                      #####
######            *scheduling.podPriority.enabled                           #####
######            **scheduling.userPlaceholder.enabled                      #####
######            **scheduling.userPlaceholder.replicas                     #####
#################################################################################
{{- end }}

{{- if eq .Values.proxy.https.enabled false }}
{{- if or (not (eq .Values.proxy.https.type "letsencrypt")) (not (eq (.Values.proxy.https.letsencrypt.contactEmail | default "") "")) }}{{ println }}
#################################################################################
######   WARNING: proxy.https.enabled is set to false by default since      #####
######            version 0.10.0. It is now set to false but proxy.https    #####

> ######            seem to have been modified indicating you may want it     #####
> ######            enabled.                                                  #####

Reviewer: ```suggestion
######            has been modified indicating you may want it enabled.     #####
```

{{- $breaking = print $breaking (include "jupyterhub.authDep.remapOldToNew" .) }}
{{- else }}
{{- $breaking = print $breaking "\n\nREMOVED: Please remove the empty 'auth' config" }}
{{- end }}
{{- end }}

{{- if hasKey .Values.proxy "containerSecurityContext" }}
{{- $breaking = print $breaking "\n\nRENAMED: proxy.containerSecurityContext has been renamed to proxy.chp.containerSecurityContext" }}
{{- end }}

{{- if hasKey .Values.proxy "pdb" }}
{{- $breaking = print $breaking "\n\nRENAMED: proxy.pdb has renamed to proxy.chp.pdb" }}
{{- end }}

{{- if hasKey .Values.proxy "networkPolicy" }}
{{- $breaking = print $breaking "\n\nRENAMED: proxy.networkPolicy has been renamed to proxy.chp.networkPolicy" }}
{{- end }}

{{- if hasKey .Values.hub "uid" }}
{{- $breaking = print $breaking "\n\nRENAMED: hub.uid must as of 1.0.0 be configured using hub.containerSecurityContext.runAsUser" }}
{{- end }}

{{- if hasKey .Values.hub "imagePullSecret" }}
{{- $breaking = print $breaking "\n\nREMOVED: hub.imagePullSecret has been removed, but there is now a chart wide wide configuration named imagePullSecret" }}
{{- end }}

{{- if hasKey .Values.singleuser "imagePullSecret" }}

> {{- $breaking = print $breaking "\n\nCHANGED: singleuser.imagePullSecret has been removed, but there is now a chart wide wide configuration named imagePullSecret" }}

Reviewer: ```suggestion
{{- $breaking = print $breaking "\n\nREMOVED: singleuser.imagePullSecret has been removed, but there is now a chart wide wide configuration named imagePullSecret" }}
```
